config/rbac/role.yaml

```diff
@@ -12,6 +12,21 @@ rules:
   - configmaps
   verbs:
   - get
+- apiGroups:
+  - ""
+  resources:
+  - configmaps
+  - events
+  - pods
+  - pods/exec
+  - secrets
+  verbs:
+  - create
+  - get
+  - list
+  - patch
+  - update
+  - watch
```

Comment:
Why are we adding these new cluster wide permissions?

Reply:
We need access to the alert manager pods in two other namespaces to exec commands within them. Is there a nicer way to do it?

Reply:
Ideally we should restrict permissions via `resourceNames`, e.g.:
https://github.com/integr8ly/integreatly-operator/blob/15f3aa133793cd859cb8630508b99f520256fb09/config/rbac/role.yaml#L30-L37 
Do we really need access to secrets, configmaps, events for pod/exec?

Why can't we do the http call from integreatly-operator using go? Why does it have to be curl executed in the alert manager pod?

Reply:
I can adjust the resources, we only need the pods/exec. 

We need to curl within the pod because alertmanager is behind a oauth proxy and it listens on 127.0.0.1/9093

Reply:
I am not 100% sure, but I think we would be able to make a call with integreatly-operator ServiceAccount token passed in `Authorization: Bearer` header. We just need to check that SA has the permissions that are checked by oauth proxy(probably view on some ns).

Reply:
I'll have a look into that and see if it's possible.

Reply:
Updated to work with Bearer auth and removed unnecessary permissions.